### ISSUE TITLE
Try CodeIntelligence's Go patches

### ISSUE THREAD

**Issue by oliverchang** [google/oss-fuzz#7164]
Link: https://github.com/CodeIntelligenceTesting/go/tree/dev.libfuzzer.18

This adds support for proper value profiling and better coverage instrumentation (8bit counters).

**Comment by oliverchang** [comment]
@Alan32Liu This could be a starter OSS-Fuzz infra project for you!

**Comment by Alan32Liu** [comment]
Thanks! That is exciting : )
Looking at it now 😀

**Comment by oliverchang** [comment]
Great! Let's be a bit more cautious here with the rollout and create a `base-builder-go-experimental` image. Then, we can turn this on project by project by setting the base image in e.g. https://github.com/google/oss-fuzz/blob/master/projects/go-coredns/Dockerfile#L17

**Comment by kyakdan** [comment]
@oliverchang , @Alan32Liu we've also updated the `go114-fuzz-build` to accept instrumentation filters so that you can limit the scope of instrumentation to the packages of interest. This proved effective to focus fuzzing on the code we want to test. Would this be interesting to integrate?

**Comment by dbrumley** [comment]
I looked through this; is this just patches to libfuzzer for golang?  Why not push them upstream to go?

**Comment by kyakdan** [comment]
> I looked through this; is this just patches to libfuzzer for golang? Why not push them upstream to go?

Yes, that is the plan indeed. We are now finalizing the patches to send for review

**Comment by 0roman** [comment]
@oliverchang I've created https://github.com/google/oss-fuzz/pull/7302 to change all go projects to use image gcr.io/oss-fuzz-base/base-builder-go-codeintelligencetesting. 

In the next step I would like to add instrumentation filters to every go project.
